### PR TITLE
feature: unit tests

### PR DESCRIPTION
**Pull request by melancholiai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,4 +55,11 @@ The program is invoked by command, a command can be successful or not. in any ca
...
     - invalid_argument: 128 - an invalid argument was given.
 
-For additional info on given errors logs are located in /var/log/cosmocrat-cli/
\ No newline at end of file
+For additional info on given errors read the stderr output stream
+
+**Tests:**
+
+run tests with
+```
```
Comment (syncush) [inline README.md]:
```suggestion
```sh
```

File: pytest.ini [hunk 2]
```diff
@@ -1,4 +1,8 @@
 [pytest]
 
 addopts = -vs
-filterwarnings = ignore::DeprecationWarning
\ No newline at end of file
+filterwarnings = ignore::DeprecationWarning
```
Comment (syncush) [inline pytest.ini]: why do you filter warnings?

File: tests/definitions.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import os
+
+EMPTY_STRING=''
+FILE_PATH='/tmp/cosmocrat-cli/data'
```
Comment (syncush) [inline tests/definitions.py:4]: is it possible that between two different runs the cli would conflict on an existing temp file? should we truncate the folder before we run the cli?

File: tests/test_osm_tools/test_osmconvert.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        assert result == mock_get_stdout.return_value
+
+    @mock.patch(f'{OSMCONVERT_MODULE_PATH}.run_command_wrapper')
+    def test_set_osm_file_timestamp(self, mock_run_command_wrapper):
+        input_path = FAKE_MULTI_PURPOSE
+        timestamp = FAKE_MULTI_PURPOSE[::-1]
+
+        with mock.patch('os.rename') as mock_rename, \
+        mock.patch(f'{OSMCONVERT_MODULE_PATH}.uuid4', return_value='uuid') as mock_uuid4, \
+        mock.patch(f'{OSMCONVERT_MODULE_PATH}.deconstruct_file_path',
```
Comment (syncush) [inline tests/test_osm_tools/test_osmconvert.py:40]: can we indent?

**Review by syncush — APPROVED** [review]

File: pytest.ini (hunk 2 above)
Comment (melancholiai) [reply pytest.ini]: it was configured by default for some reason, but in the documentation I see it is not recommended.
anyway, I've removed the filter.

File: README.md (hunk 1 above)
Comment (melancholiai) [reply README.md]: done.

File: tests/definitions.py (hunk 3 above)
Comment (melancholiai) [reply tests/definitions.py:4]: functionality wouldn't break unless the user will use multiple cli instances to run commands that will conflict with each other at the same time.
once a command has finished the temp files are not needed.

File: tests/test_osm_tools/test_osmconvert.py (hunk 4 above)
Comment (melancholiai) [reply tests/test_osm_tools/test_osmconvert.py:40]: it's a long `with` statement, no need for indentation

**Pull request merged**